lib/developer_portal/app/controllers/developer_portal/admin/account/personal_details_controller.rb

```diff
@@ -49,16 +45,17 @@ def deny_unless_can_update
     end
   end
 
-  def redirect_path
-    if params[:origin] == "users"
-      if current_account.provider?
-        provider_admin_account_users_path
-      else
-        admin_account_users_path
-      end
-    else
-      admin_account_personal_details_path
-    end
+  def user_params
+    params.require(:user).permit([:current_password] +
+                                   resource.special_fields +
+                                   resource.defined_fields.map(&:name))
   end
 
+  def verify_current_password
+    return unless current_user.using_password?
+    return if current_user.authenticated?(user_params[:current_password])
+
+    flash.now[:error] = 'Current password is incorrect.'
+    resource.errors.add(:current_password, 'Current password is incorrect.')
```

Comment:
```suggestion
    resource.errors.add(:current_password, 'is incorrect')
    flash.now[:error] = resource.errors.full_messages.to_sentence
```

This outcome of this is slightly different (see screenshots below), but I think it is actually more standard - showing only a short error message on the input hint (which does not duplicate the field name), and then show the full message on the flash.

While the current templates only check for the specific errors (e.g. for `password` and `current_password` fields), I think we need to make sure that `resource.errors.full_messages.to_sentence` actually returns something meaningful. But with the current approach it would return `"Current password Current password is incorrect."`.

The suggested option still doesn't use I18n, but I think in this case it would have to be:
```
resource.errors.add(:current_password, t('activerecord.errors.models.user.current_password_incorrect'))
```

and adding `current_password_incorrect: 'is incorrect'` on the corresponding path of `en.yml`, because `:current_password` is not an actual field of User, so we can't treat it as a normal "attribute error".
This is not too elegant, but maybe could be done also.

And the screenshots... 

**Before:**
![Screenshot from 2023-10-02 15-49-07](https://github.com/3scale/porta/assets/1270328/8a31b6a5-6cb1-479e-afda-0d0e2416da6c)

**After:**
![Screenshot from 2023-10-02 15-48-37](https://github.com/3scale/porta/assets/1270328/148cc764-7068-40ef-8035-192a4c175d03)

As I said before, the "After" style is more similar to other fields' validation errors:

![Screenshot from 2023-10-02 15-47-27](https://github.com/3scale/porta/assets/1270328/3ca4b4ca-336e-4b94-b3fd-054b4425511c)
![Screenshot from 2023-10-02 15-47-46](https://github.com/3scale/porta/assets/1270328/6660a224-6231-412e-a555-541dacb97099)